src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -55,11 +56,37 @@ public NodeSettings(Network innerNetwork = null, ProtocolVersion protocolVersion
             this.LoggerFactory.AddConsoleWithFilters();
             this.LoggerFactory.AddNLog();
             this.Logger = this.LoggerFactory.CreateLogger(typeof(NodeSettings).FullName);
+
+            // TODO: Temporary until Build method updated to call LoadConfiguration
```

Comment:
So the next step is that LoadConfiguration will be called from the Build method, right?
Effectively LoadConfiguration will be called in a loop for every feature?
If it's the case then can we remove the `loadConfiguration = true` from the ctor params and just call LoadConfiguration at the end of the ctor?

Reply:
Correct regarding the loop. However NodeSettings may be constructed at a different time than its LoadConfiguration is called so we do need that parameter to suppress LoadConfiguration in the constructor.

Reply:
> NodeSettings may be constructed at a different time than its LoadConfiguration is called

I'm not sure what you mean by that. NodeSettings is constructed, then we do something else, then we call LoadConfiguration on NodeSettings?
I'd prefer it if you removed this TODO, removed the `loadConfiguration = true` from the parameters so that in the next PR you can start with this properly. Here it feels like you're doing a little bit of the next PR (and it's confusing because `loadConfiguration:false` gets overwritten).

Reply:
I'm not sure if you noticed but I agreed with your above comment after initially disagreeing in our PMs (which you are quoting here). When I responded to the PM I was not entirely clear what you meant.

If we try to remove loadConfiguration = true in this PR then we are adding changes in another +- 20 files to it. It makes more sense that this change should go together with adding the LoadConfiguration call to the Build method - which I am hoping we can do in the next PR - in the interest if keeping this PR small.